portfolio/src/main/webapp/script.js

```diff
@@ -31,11 +31,27 @@ function getFunFact() {
 }
 
 /**
- * Fetches a the message at /data.
+    Gets and displays my bottled water rankings from /data.
  */
-async function loadGreetingsContainer() {
-  console.log('Fetching data message.');
-  const response = await fetch('/data');
-  const quote = await response.text();
-  document.getElementById('server-message-container').innerText = quote;
-}
\ No newline at end of file
+ async function getWaterRankings() {
+     console.log('Fetching water ratings.');
+     const response = await fetch('/data');
+     const json = await response.json();
+     const ranking_container = document.getElementById('water-ranking-container');
+     ranking_container.innerText = '';
+     if(ranking_container.children.length == 0) {
```

Comment:
instead of this if condition, i think you should reset the list of children and refetch the new entries from your servlet.

Reply:
Done